src/Clients/CaradhrasCardClient.php

```diff
@@ -0,0 +1,469 @@
+use Idez\Caradhras\Exceptions\CaradhrasException;
+use Idez\Caradhras\Exceptions\CVVMismatchException;
+use Idez\Caradhras\Exceptions\FailedRequestCardBatchException;
+use Idez\Caradhras\Exceptions\FindCardsException;
+use Idez\Caradhras\Exceptions\GetCardDetailsException;
+use Idez\Caradhras\Exceptions\IssuePhysicalCardException;
+
+class CaradhrasCardClient extends BaseApiClient
+{
+    public const CARD_NOT_PRINTED = 'Invalid card status for activation.';
+
+    /**
+     * Associate card to account.
+     *
+     * @param int $cardId
+     * @param int $accountId
+     * @param int $individualId
+     * @return bool
+     * @throws RequestException
+     */
+    public function associatePrepaidCardToAccount(int $cardId, int $accountId, int $individualId): bool
+    {
+        $this->apiClient()
+            ->post("/contas/{$accountId}/atribuir-cartao-prepago", [
+                'idCartao' => $cardId,
+                'idPessoaFisica' => $individualId,
+            ])
+            ->throw();
+
+        return true;
+    }
+
+    /**
+     * @param  int  $cardId
+     * @return Card
+     */
+    public function unlockSystemBlockedCard(int $cardId): Card
+    {
+        $unlockRequest = $this->apiClient()->post("/cartoes/{$cardId}/desbloquear-senha-incorreta");
+
+        return new Card($unlockRequest);
+    }
+
+    /**
+     * Update card password.
+     *
+     * @param  string  $cardId
+     * @param  string  $password
+     * @return array|object
+     */
+    public function updateCardPassword(string $cardId, $password)
+    {
+        return $this->apiClient()
+            ->withHeaders(['senha' => str_pad($password, 4, '0', STR_PAD_LEFT)])
+            ->put("/cartoes/{$cardId}/alterar-senha")
+            ->object();
+    }
+
+    /**
+     * Issue physical card.
+     *
+     * @param  int  $accountId
+     * @param  int  $individualId
+     * @return Card
+     * @throws \Illuminate\Http\Client\RequestException
+     * @throws \Exception
+     */
+    public function issuePhysicalCard(int $accountId, int $individualId): Card
+    {
+        $data = [
+            'id_pessoa' => $individualId,
+            'id_tipo_plastico' => config('app.plastic_id'),
+        ];
+
+        $issueCardResponse = $this->apiClient(false)
+            ->post("/contas/{$accountId}/gerar-cartao-grafica", $data);
+
+        if ($issueCardResponse->failed()) {
+            throw new IssuePhysicalCardException();
+        }
+
+        $cardId = $issueCardResponse->json('idCartao');
+
+        $this->setCardPassword($cardId, random_int(1000, 9999));
+
+        return $this->getCard($cardId);
+    }
+
+    /**
+     * Create virtual card.
+     *
+     * @param  int  $accountId
+     * @param  int  $individualId
+     * @return Card
+     */
+    public function issueVirtualCard(int $accountId, int $individualId): Card
+    {
+        $data = http_build_query([
+            'dataValidade' => now()->addYears(5)->toDateString() . 'T00:00:00.000',
```

Comment:
Poderiamos passar isso aqui por Parametro, e esse formato de data tem um formatação no Carbon que já faz isso.

```suggestion
            'dataValidade' => now()->addYears(5)->format('Y-m-d\TH:i:s')
```

Reply:
isso o que?

Reply:
a formtação da data.